src/App.js

```diff
@@ -3,28 +3,50 @@ import React, { useState, useEffect } from "react";
 // import {TodoList} from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 import TodoList from "./TodoList";
-
 // import React from "react";
 
+function App() {
 
-function useSemiPersistentState() {
-  // console.log(JSON.parse(localStorage.getItem("todoList")))
-  const TodoList = localStorage.getItem("todoList")
-    ? JSON.parse(localStorage.getItem("todoList"))
-    : [];
-  
-  const [todoList, setTodoList] = useState(
-   TodoList
-  );
+const [todoList, setTodoList] = useState(
+  localStorage.getItem("todoList")
+  ? JSON.parse(localStorage.getItem("todoList"))
+  : []
+  // cuando tiene mayusculas son componentes y minsculas variables,state
+);
 
-  useEffect(() => {
-    localStorage.setItem("todoList", JSON.stringify(todoList));
-  }, [todoList]); //dependent lo que hace es buscar.
-  return [todoList, setTodoList];
-}
+const [isLoading,setIsLoading]=useState(true);
+
+useEffect(() => {
+  new Promise ((resolve, reject) => {
+    
+  setTimeout(()=>{ //establece un temporizador que ejecuta una función o una pieza de código 
+    resolve({
+      data:{
+       todoList:
+    JSON.parse(localStorage.getItem("todoList")),
+             },
+           });
+           console.log('2 second has passed')
+         }, 2000); //tiempo de retardo
+          }
+      ).then((result) => { //método de un Promiseobjeto toma hasta dos argumentos: funciones de devolución de llamada para los casos cumplidos y rechazados del Promise
+      console.log(result)
+      setTodoList([...result.data.todoList])
+      setIsLoading(false); 
+      // isLoading(false);
+      console.log(result);
+  });
+}, []);
+
+
+useEffect(() => {
+
+if (isLoading) {
```

Comment:
If you run the app, input some items in the todolist and refresh page - the list items no longer persist : (. This function doesn't quite work as intended! Mind fixing that?